src/test/java/com/example/springbootbookshop/controller/BookControllerTest.java

```diff
@@ -0,0 +1,232 @@
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext,
+                          @Autowired DataSource dataSource) throws SQLException {
+        mockMvc = MockMvcBuilders.webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        tearDown(dataSource);
+    }
+
+    @BeforeEach
+    void beforeEach(@Autowired DataSource dataSource) throws SQLException {
+        tearDown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/category/insert-category.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/insert-books.sql"));
+        }
+    }
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        tearDown(dataSource);
+    }
+
+    @SneakyThrows
+    static void tearDown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/category/clear-category.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/clear-books-db.sql"));
+        }
+    }
+
+    @Test
+    @DisplayName("Create a new book")
+    @Sql(scripts = "classpath:database/books/clear-books-db.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    void createNewBookAndSaveIntoDbAssertSuccess() throws Exception {
+        CreateBookRequestDto createBookRequestDto = new CreateBookRequestDto(
+                "Book1",
+                "Author1",
+                "ISBN1",
+                BigDecimal.TEN,
+                "Description1",
+                "coverImageUrl1",
+                List.of(1L));
+        BookDto expectedDto = getExpectDto(createBookRequestDto);
```

Comment:
You don't need to create expectedDto here.
In the end of the test you should compare createBookRequestDto with actualDto